corehq/ex-submodules/casexml/apps/case/tests/util.py

```diff
@@ -29,7 +29,16 @@ def __init__(self, xml_element, version=V2):
         self.version = version
 
     def to_string(self):
-        return ElementTree.tostring(self.xml_element)
+        restore_element_copy = ElementTree.fromstring(ElementTree.tostring(self.xml_element))
+        # hack need to explicitly convert date_opened from datetimes to dates in restores
+        # because the mobile can't handle datetimes.
+        # see: http://manage.dimagi.com/default.asp?230043 for more info
+        date_opened_element = restore_element_copy.find(
+            '{http://commcarehq.org/case/transaction/v2}update/{http://commcarehq.org/case/transaction/v2}date_opened'
+        )
+        if date_opened_element is not None and date_opened_element.text:
+            date_opened_element.text = date_opened_element.text[:10]
```

Comment:
Is this because all the tests initialize the date_opened as a date time? I think you should just be able to change this whenever a `CaseBlock` gets initialized. Otherwise I feel like this is going to get really confusing down the line.